bucketlistapi/resources.py

```diff
@@ -0,0 +1,323 @@
+            # implement pagination for name search or bucketlists for user
+            limit = args.get('limit', 20) if (
+                args.get('limit', 20) < 100) else 100
+            page = args.get('page', 1)
+            name_search = args.get('q')
+            bucketlists = g.user.bucketlists
+            if name_search:
+                bucketlists = bucketlists.filter(
+                    BucketList.name.ilike("%{}%".format(name_search.title())))
+            if not bucketlists.all():
+                abort(
+                    404, message='no bucketlist containing {}'
+                    .format(name_search))
+            bucketlists = bucketlists.paginate(
+                page=page, per_page=limit, error_out=False)
+            prev_page = str(request.url_root) +\
+                'api/v1/bucketlists?' + 'limit=' + str(limit) + '&page=' +\
+                str(page - 1) if bucketlists.has_prev else None
+            next_page = str(
+                request.url_root) + 'api/v1/bucketlists?' + 'limit=' +\
+                str(limit) + '&page=' + str(page + 1) if (
+                bucketlists.has_next) else None
+            bucketlists_per_page = OrderedDict([('Bucketlist{}'.format(
+                bucketlist.id), bucketlist.as_dict())
+                for bucketlist in bucketlists.items])
+            if bucketlists_per_page:
+                return {
+                    "data": bucketlists_per_page,
+                    "pages": bucketlists.pages,
+                    "previous_page": prev_page,
+                    "next_page": next_page
+                }, 200
+            return abort(404, message='You don\'t have any bucketlist yet!')
+
+    @use_args(name_field)
+    def put(self, args, bucketlist_id=None):
+        # update a bucketlist with id(bucketList_id)
+        if not bucketlist_id or not (
+                BucketList.query.get(
+                    bucketlist_id).created_by == self.created_by):
+            abort(400, message="method not supported for"
+                  " the URL or invalid bucketlist id.")
+        self.name = args.get('name')
+        if not self.check_name():
+            abort(400, message="name field cannot be empty or too short")
+        elif self.bucket_list_name_exists():
+            abort(400, message='Bucket List name already exists')
+        elif db.session.query(
+            BucketList).filter(and_(
+                BucketList.created_by == self.created_by,
+                BucketList.id == bucketlist_id)).update(
+                {'name': self.name.strip().title()}):
+            db.session.commit()
+            return BucketList.query.get(bucketlist_id).as_dict()
+        abort(400, message='Could not update bucketlist {}'.format(
+            bucketlist_id))
+
+    def delete(self, bucketlist_id=None):
+        # view to delete a bucket list with the associated id
+        if not bucketlist_id:
+            abort(404, message="method not supported for the URL")
```

Comment:
The ideal status code for method not supported for a url is `405` and not `404`. Consider changing this.